dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/deploy/DefaultApplicationDeployer.java

```diff
@@ -899,12 +900,16 @@ private DynamicConfiguration getDynamicConfiguration(URL connectionURL) {
     private void registerServiceInstance() {
         try {
             registered = true;
-            MetricsEventBus.post(RegistryEvent.toRegisterEvent(applicationModel),
-                () -> {
-                    ServiceInstanceMetadataUtils.registerMetadataAndInstance(applicationModel);
-                    return null;
-                }
-            );
+            List<ServiceDiscovery> serviceDiscoveries = ServiceInstanceMetadataUtils.getServiceDiscoveries(applicationModel);
+            if (serviceDiscoveries.size() > 0) {
+                MetricsEventBus.post(RegistryEvent.toRegisterEvent(applicationModel, ServiceInstanceMetadataUtils.getServiceDiscoveryNames(serviceDiscoveries)),
```

Comment:
Move this `MetricsEventBus` into `ServiceInstanceMetadataUtils.registerMetadataAndInstance` would be better

Reply:
ok